src/KafkaFlow/Consumers/ConsumerContext.cs

```diff
@@ -2,50 +2,77 @@ namespace KafkaFlow.Consumers
     {
+        private readonly TaskCompletionSource<TopicPartitionOffset> completionSource = new();
         private readonly IConsumer consumer;
         private readonly IOffsetManager offsetManager;
-        private readonly ConsumeResult<byte[], byte[]> kafkaResult;
+        private readonly IConsumerWorker worker;
+        private readonly IDependencyResolverScope messageDependencyScope;
 
         public ConsumerContext(
             IConsumer consumer,
             IOffsetManager offsetManager,
             ConsumeResult<byte[], byte[]> kafkaResult,
-            CancellationToken workerStopped,
-            int workerId)
+            IConsumerWorker worker,
+            IDependencyResolverScope messageDependencyScope,
+            IDependencyResolver consumerDependencyResolver)
         {
-            this.WorkerStopped = workerStopped;
-            this.WorkerId = workerId;
+            this.ConsumerDependencyResolver = consumerDependencyResolver;
             this.consumer = consumer;
             this.offsetManager = offsetManager;
-            this.kafkaResult = kafkaResult;
+            this.worker = worker;
+            this.messageDependencyScope = messageDependencyScope;
+            this.TopicPartitionOffset = new TopicPartitionOffset(
+                kafkaResult.Topic,
+                kafkaResult.Partition.Value,
+                kafkaResult.Offset.Value);
+            this.MessageTimestamp = kafkaResult.Message.Timestamp.UtcDateTime;
         }
 
         public string ConsumerName => this.consumer.Configuration.ConsumerName;
 
-        public CancellationToken WorkerStopped { get; }
+        public CancellationToken WorkerStopped => this.worker.StopCancellationToken;
 
-        public int WorkerId { get; }
+        public int WorkerId => this.worker.Id;
 
-        public string Topic => this.kafkaResult.Topic;
+        public IDependencyResolver WorkerDependencyResolver => this.worker.WorkerDependencyResolver;
 
-        public int Partition => this.kafkaResult.Partition.Value;
+        public IDependencyResolver ConsumerDependencyResolver { get; }
 
-        public long Offset => this.kafkaResult.Offset.Value;
+        public string Topic => this.TopicPartitionOffset.Topic;
+
+        public int Partition => this.TopicPartitionOffset.Partition;
+
+        public long Offset => this.TopicPartitionOffset.Offset;
+
+        public TopicPartitionOffset TopicPartitionOffset { get; }
 
         public string GroupId => this.consumer.Configuration.GroupId;
 
         public bool ShouldStoreOffset { get; set; } = true;
 
-        public DateTime MessageTimestamp => this.kafkaResult.Message.Timestamp.UtcDateTime;
+        public DateTime MessageTimestamp { get; }
 
-        public void StoreOffset() => this.offsetManager.MarkAsProcessed(this.kafkaResult.TopicPartitionOffset);
+        public Task<TopicPartitionOffset> Completion => this.completionSource.Task;
+
+        public void StoreOffset()
```

Comment:
Change the name to Complete()

Reply:
Changed. Also changed the Configuration method from `WithManualStoreOffsets` to `WithManualMessageCompletion` and the `ConsumerContext` property from `ShouldStoreOffset` to `AutoCompleteMessage`